### ISSUE TITLE
ARG: Add option to enforce granularity

### ISSUE THREAD

**Issue by ofek** [astral-sh/ruff#8503]
It would be nice if there was [an option](https://docs.astral.sh/ruff/settings/#flake8-unused-arguments) for the ARG rules that would force ignore comments to apply just to the particular unused arguments. Often I want to take note in the code that one or two are intentionally ignored but the others should definitely not be. When this option exists there should be another rule added that would allow you to actually ignore the entire function/method.

Example from https://github.com/pypa/hatch/pull/1031:

```
def build_wheel(
    wheel_directory: str,
    config_settings: dict[str, Any] | None = None,  # noqa: ARG001
    metadata_directory: str | None = None,  # noqa: ARG001
) -> str:
```

**Comment by charliermarsh** [comment]
Can you expand on this a bit, or give an example of what you mean? (Do the ignore comments in your example above not work as expected?)

**Comment by ofek** [comment]
No I mean I want to enforce the example from above rather than allowing the comment to apply to the entire function

**Comment by charliermarsh** [comment]
If I plug that into the [playground](https://play.ruff.rs/f7660caa-b0db-46ba-bba7-6b5bfc3572e3), I see that the second two arguments _are_ ignored, while the first is flagged with `ARG001`. I'm really sorry because I suspect I'm still just misunderstanding the ask!

**Comment by charliermarsh** [comment]
Ahh got it, thanks!
